### ISSUE TITLE
Is a non-blocking version of this module planned?

### ISSUE THREAD

**Issue by ghost** [mdirolf/nginx-gridfs#30]
Hi,

Do you plan on producing a non-blocking version of this module?  Performance would be much better if it were non-blocking (which would require the C library to be able to be used asynchronously too, which last time I looked, it wasn't).

Thanks,

Marcus.

**Comment by banker** [comment]
There's a issue for adding non-blocking support. Once we add that, we can look at make it non-blocking in the module:
https://jira.mongodb.org/browse/CDRIVER-27

**Comment by ghost** [comment]
Hi Kyle,

Thanks for the reply.  Is there any timeframe wrt adding async support to the C client?  I may well be interested in getting involved in the devel work for a non-blocking Nginx module for Mongo and GridFS, but will obviously wait for the non-blocking client to be developed first.

Btw, I'm not clear what the issue mentioned in the link you posted is, as it doesn't have any publicly viewable info as far as I can tell.

Cheers.

**Comment by banker** [comment]
Within two months, max.

**Comment by ghost** [comment]
Thanks.

**Comment by ghost** [comment]
Hi everyone, 
Is there any updates on the issue? Two months passed, I'm not pushing of course, but the module is just a proof of concept without async access. We consider GridFS, but still have to store files in the filesystem with simple filename links from Mongo and maintain a complicated backup scheme. GridFS would really be a welcome addition to the stack.
Thanks!

**Comment by reviyou** [comment]
Yes we are very interested what creators of this module think of this issues - we did some research and it looks like this is the only single blocking issue for many people.
I wonder what performance limitations\degradations it might cause but what's more important - what workaround we can implement long term(even if we can rewrite part of this module by ourselves) - it is dependant on Mongo team to resolve their issue? Is it even possible, etc..

It scares a bit that there was no update for 3 years and we might we looking at the old outdated solution nobody uses anymore.. but what others options should we look at?

**Comment by lloydzhou** [comment]
mongo-c-driver already released v1.0.2  https://github.com/mongodb/mongo-c-driver/releases/tag/1.0.2, this version is a no-blocking driver.
this project has using the legacy version...
